python/docs/logger.md

```diff
@@ -45,8 +45,23 @@ In the BeeAI framework, the `Logger` class is an abstraction built on top of Pyt
 
 To use the logger in your application:
 
+<!-- embedme examples/logger/base.py -->
 ```py
-# Coming soon
+import logging
+
+from beeai_framework.utils import BeeLogger
+
+# Configure logger with default log level
+logger = BeeLogger("app", level=logging.TRACE)
+
+# Log at different levels
+logger.trace("Trace!")
```

Comment:
Can we add typings for a custom level? My IDE complains that there is no such method.

Reply:
I did some looking into this and Mypy just doesn't support `setattr` since it is static analysis and `setattr` is dynamically set at runtime. So in this case we would want to just exempt this from Mypy, but adding a Mypy ignore in an example that's embedded in the docs would look bad.

Reply:
we can just add such method to the BeeLogger

Reply:
As far as I found online that's non-trivial and I believe the dynamic solution we're using is actually the recommend way of adding logging levels. If want to revisit it I would suggest we open up a follow up issue and merge this as is since it works as intend despite the Mypy complaints

Reply:
I meant this.

```python
from typing import TYPE_CHECKING

class BeeLogger(logging.Logger):
    if TYPE_CHECKING:
        trace = logging.Logger.debug
```

Reply:
This satisfies MyPy without any modifications.